db/models/Additive.js

```diff
@@ -0,0 +1,15 @@
+import mongoose from "mongoose";
+
+const { Schema } = mongoose;
+
+const additiveSchema = new Schema({
+  name: { type: String, required: true },
+  id: { type: String, required: true },
+  userId: { type: String, required: true },
+});
+additiveSchema.index({ userId: 1, id: 1 }, { unique: true });
```

Comment:
can you explain to me what this line of code is doing? I don't understand :D

Reply:
https://www.mongodb.com/docs/manual/core/index-compound/


It is to avoid double saves of the same profuct but when you have different Users you should be able to save the same thing more than once in the database

Reply:
thanks, that's cool